src/interpreter/ByteCodeGenerator.cpp

```diff
@@ -269,7 +260,7 @@ void ByteCodeGenerator::collectByteCodeLOCData(Context* context, InterpretedCode
 
     // Parsing
     Node* ast = nullptr;
-    if (codeBlock->isGlobalScope()) {
+    if (codeBlock->isGlobalScope() || codeBlock->isEvalCode()) {
```

Comment:
Without this, backtrace in `eval("g = 2\ng = 3")` enters an infinite loop. I don't know if this fix is correct.

Reply:
We need to parse the entire source code for global code or eval code.
So, please fix this line as below
```c++
if (codeBlock->isGlobalCodeBlock()() || codeBlock->isEvalCode()) {
```

Reply:
Thank you, updated.